### PR TITLE
Addition of Area2D to Mob as a child

### PR DESCRIPTION
**Pull request by AleksandrHovhannisyan**
To be used for collision detection. Will work on incorporating Area2D + CollisionShape2D into the other AnimatedEntity scenes as well.

### CODE REVIEW

**Comment by AleksandrHovhannisyan** [comment]
Added some changes. I'd merge at this point.

**Comment by AleksandrHovhannisyan** [comment]
Okay, so maybe we should first merge Matt's, and then I'll do a pull rebase and make another PR?

File: environment_scenes/World.gd [hunk 1]
```diff
@@ -4,43 +4,46 @@ extends Node2D
...
+	
+	
+	
+# Updates the on-screen HUD mana bar to reflect player's current mana
+func update_mana_bar():
+	$Canvas/HUD/Mana.value = $Player.mana
+	
+	
+	
+func update_HUD_bars():
```
Comment (RustyBamboo) [inline environment_scenes/World.gd]: good

File: entity_scenes/Mob.gd [hunk 2]
```diff
@@ -11,13 +13,16 @@ func _ready():
...
-		
-	
 	move(velocity)
-	pass
\ No newline at end of file
+	pass
+
+
+
+func _on_Area2D_body_entered( body ):
```
Comment (RustyBamboo) [inline entity_scenes/Mob.gd]: we'll need to remember to remove this debug

**Pull request merged**